### PR TITLE
feat: powershell target

### PR DESCRIPTION
**Pull request by vaind**
This adds powershell publishing support

### CODE REVIEW

File: src/utils/version.ts [hunk 1]
```diff
@@ -89,6 +89,8 @@ export function versionGreaterOrEqualThan(v1: SemVer, v2: SemVer): boolean {
     return true;
   } else if (v1.pre && !v2.pre) {
     return false;
+  } else if (v1.pre && v2.pre && v1.pre === v2.pre) {
+    return v1.build === v2.build;
```
Comment (asottile-sentry) [inline src/utils/version.ts:93]: I don't think this is correct for this function -- it's meant to implement `>=` so only checking strict equality isn't quite right

File: src/utils/version.ts [hunk 2]
```diff
@@ -157,7 +159,6 @@ export function getPackageVersion(): string {
  * Returns the stringified version of the passed SemVer object.
  */
 export function semVerToString(s: SemVer) {
-  return `${s.major}.${s.minor}.${s.patch}${s.pre ? `-${s.pre}` : ''}${
-    s.build ? `+${s.build}` : ''
-  }`;
+  return `${s.major}.${s.minor}.${s.patch}${s.pre ? `-${s.pre}` : ''}${s.build ? `+${s.build}` : ''
```
Comment (asottile-sentry) [inline src/utils/version.ts:162]: craft is going to have some other problems with this -- currently we don't allow `+` releases anywhere -- is this necessary?

File: Dockerfile [hunk 3]
```diff
@@ -65,6 +65,7 @@ RUN curl -fsSL https://packages.microsoft.com/config/debian/10/packages-microsof
   && cargo --version \
   && cargo install cargo-hack \
   && gem install -g --no-document \
+  && gem cleanup all \
```
Comment (asottile-sentry) [inline Dockerfile:68]: let's decouple this from your change and do it separately

**Review by asottile-sentry — COMMENTED** [review]
can you write some tests for your new feature

File: src/utils/version.ts (hunk 2 above)
Comment (vaind) [reply src/utils/version.ts:162]: this is a craft version check. I've hit this when testing the compiled craft locally - it comes with version `1.11.0-dev.0` and even though you specify that as `minVersion` in `.craft.yml`, the check fails without the changes I've made here.

**Comment by bruno-garcia** [comment]
Thanks @vaind for raising this!

**Comment by bruno-garcia** [comment]
@asottile-sentry please let me know if we need any other change to get this through

**Review by asottile-sentry — APPROVED** [review]
apologies I was out sick the last week (still am) but going to approve to unblock

**Comment by vaind** [comment]
Thx.  Can you please merge as well? I don't have permissions

**Comment by bruno-garcia** [comment]
Thanks @vaind and @asottile-sentry !

**Pull request merged**